SRVKS-1243: Add Serving transport encryption docs (#82435)
openshift/openshift-docs: modules/serving-transport-encryption-choice-of-a-certificate-issuer.adoc

:_mod-docs-content-type: CONCEPT
[id="serving-transport-encryption-choice-of-a-certificate-issuer_{context}"]
= Choice of a certificate issuer

> [NOTE]
> ====
> Issuers refer to cert-manager `Issuers` and `ClusterIssuers`. They represent certificate authorities (CAs) that can generate signed certificates by honoring certificate signing requests.
> ====

Reviewer: ```suggestion
Issuers refer to cert-manager issuers and cluster issuers. They represent certificate authorities (CAs) that can generate signed certificates by honoring certificate signing requests.
```
I don't think that this really needs to be a note. Also, formatting of these components should be consistent with other product docs:
- "issuers", per https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift#cert-manager-issuer-types_cert-manager-operator-about
- "cluster issuers" per https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift#cert-manager-acme-http01_cert-manager-operator-issuer-acme

> To identify your certificate issuer, use `cert-manager` documentation, which has examples for the following:
> 
> * A custom CA stored in a K8s secret
> * HTTP-01 challenges, e.g. Let’s encrypt
> * DNS-01 challenges
> * Self-signed issuers
> 
> [NOTE]
> ====
> Not all issuer types work for each Knative Serving encryption feature.
> ====
> 
> Depending on the encryption part you use, {ServerlessProductName} requires your certificate issuer to be able to sign certain certificates.

Reviewer: ```suggestion
Depending on the encryption features that you use, {ServerlessProductName} requires your certificate issuer to be able to sign certain certificates. To identify your certificate issuer, refer to the `cert-manager` documentation, which contains examples for the following:

* A custom CA stored in a Kubernetes secret
* HTTP-01 challenges
* DNS-01 challenges
* Self-signed issuers

== Add heading

Not all issuer types work for each Knative Serving encryption feature.

```
Instead of having the "Not all issuer types work for each Knative Serving encryption feature." part as a note here, I would add a heading to define this as a specific section, maybe something like "Supported issuer types for Knative Serving encryption features"

> * For cluster-local encryption, the issuer must be able to sign certificates for cluster-local domains of these types:

Reviewer: ```suggestion
* For cluster-local encryption, the issuer must be able to sign certificates for the following cluster-local domain types:
```

+
--
** `myapp.<namespace>`
** `myapp.<namespace>.svc`
** `myapp.<namespace>.svc.cluster.local`
--
+

> As the CA usually is not within the cluster, verification using the ACME protocol (DNS01/HTTP01) is not possible. You can use an issuer that allows creating these certificates (e.g. cert-manager’s CA issuer).

Reviewer: ```suggestion
As the CA usually is not within the cluster, verification using the ACME protocol (DNS01/HTTP01) is not possible. You can use an issuer that allows creating these certificates, such as the cert-manager CA issuer.
```
Do not use parentheses and "e.g." in docs generally. Also, I've noticed that "cert-manager" seems to be inconsistently formatted - should it be "cert-manager", "`cert-manager`", something else? Check that this is referred to consistently and is in line with whatever the cert manager docs use.

* For system-internal encryption, the issuer must be able to sign certificates with the following Subject Alternative Names (SAN):
+
--
** `kn-routing`
** `kn-user-<namespace>`, where `<namespace>` is each namespace where Knative Services are created
** `data-plane.knative.dev`
--
+

> These SANs are needed for Knative to verify connections between the internal components. As this is also not possible using the ACME protocol (DNS01/HTTP01), you need to configure an issuer that allows creating these certificates, for example, `cert-manager` CA issuer.

Reviewer: See previous note about inconsistent formatting for cert-manager - make these consistent.

[id="additional-resources_serving-transport-encryption-choice-of-a-certificate-issuer"]
[role="_additional-resources"]
== Additional resources

* link:https://cert-manager.io/docs/concepts/issuer/[cert-manager documentation on issuers]

> * link:https://cert-manager.io/docs/configuration/issuers/[List of cert-manager integrations]

Reviewer: Are there downstream docs available for this considering https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift ?

Author: None of the cert-manager.io links that are in this PR have a direct counterpart in the downstream docs.
